newsroom/agenda/formatters/csv_formatter.py

```diff
@@ -0,0 +1,99 @@
+from newsroom.formatter import BaseFormatter
+import csv
+import io
+
+
+class CSVFormatter(BaseFormatter):
+    VERSION = "1.0"
+    PRODID = "Newshub"
+    FILE_EXTENSION = "csv"
+    MIMETYPE = "text/csv"
+
+    def format_item(self, item, item_type=None):
+        event_item = self.format_event(item)
+        return self.serialize_to_csv(event_item)
+
+    def serialize_to_csv(self, data):
+        csv_string = io.StringIO()
+        csv_writer = csv.DictWriter(csv_string, delimiter=",", fieldnames=data.keys())
+        csv_writer.writeheader()
+        csv_writer.writerow(data)  # Write a single row for the provided data
+        csv_string.seek(0)  # Reset the buffer position
+        return csv_string.getvalue().encode("utf-8")
+
+    def format_event(self, item):
+        event = item.get("event", {})
+        return {
+            "Event name": item.get("name", ""),
+            "Description": item.get("definition_long", item.get("definition_short", "")),
+            "Language": item.get("language", ""),
+            "Event start date": self.format_date(item, "start"),
+            "Event end date": self.format_date(item, "end"),
+            "Event time": f"{self.format_time(item, 'start')}-{self.format_time(item, 'end')}",
+            "Event timezone": item.get("dates", {}).get("tz", ""),
+            "Location": self.parse_location(item, "name"),
+            "Country": self.parse_location(item, "country"),
+            "Subject": self.parse_list(event, "subject"),
+            "Website": event.get("links")[0] if event.get("links") else "",
+            "Category": self.parse_list(event, "anpa_category"),
+            "Event type": item.get("item_type", ""),
+            "Organization name": item.get("event", {}).get("event_contact_info", {})[0].get("organisation", " "),
+            "Contact": self.parse_contact_info(item),
+            "Coverage type": self.parse_coverage(item, "coverage_type"),
+            "Coverage status": self.parse_coverage(item, "coverage_status"),
+        }
+
+    def format_date(self, item, date_type):
+        date_obj = item.get("dates", {}).get(date_type)
+        if date_obj:
+            return date_obj.strftime("%Y-%m-%d")
+        return ""
+
+    def format_time(self, item, date_type):
```

Comment:
I think we need to handle also events without end time and all day events

Reply:
I see, so If event is `all_day = true` in this case we have to add 'all_day' in  `event time `
and if no `end_time` then show only `start time`  right ?

Reply:
I guess we can keep it empty when it's all day, for `no_end_time` only show start time

Reply:
Okay Got it